src/resource_pool_texture.cpp

```diff
@@ -7,28 +7,33 @@ namespace wr
 
 	TexturePool::TexturePool()
 	{
+#ifdef _DEBUG
+		static uint16_t pool_count = 0u;
```

Comment:
The render system should be in charge of creating pools. You could make the constructor of pool private and friend it with render system. so you can ONLY create it using the RS. that way you can count the number of pools from that list.,